### DISCUSSION TITLE
Cover image do not respect image filename(capital letter) in Settings

### DISCUSSION THREAD

**Discussion by Emilot** [jcorporation/myMPD#863] [Q&A] [answered]
myMPD version: [latest]

**Describe the bug**

If i put Cover as the name of the cover filename in Settings, will be ignored by the myMPD. If i wrote cover (lower case) then it appears. The same happen if i put Front, instead of front and anything else.

**To Reproduce**

1. Create a new album folder
2. put a cover image
3. Name it Cover.jpg
4. Update your Library
5. Not showing

**Server plattform (please complete the following information):**
 - MPD version: [0.24]

**Client plattform (please complete the following information):**
 - Browser Firefox latest

**Debug logs (please attach if it can be usefull)**

Look at https://jcorporation.github.io/myMPD/debug
No errors

**Comment by jcorporation** [comment]
I am unsure if I understand this issue correctly.

Please post 
- the exact not working setting
- filesystem listing of the folder with the not working coverimage 
- the debug logs

Generally the coverimage handling is case sensitive.

**Comment by Emilot** [comment]
Clarification

cover.jpg works
Cover.jpg do not work

Sorry for the confusion text.

**Comment by jcorporation** [comment]
debug logs please.

**Comment by Emilot** [comment]
If i edit the Cover.jpg filename and change it to cover.jpg it shows as expected. If i rename it again to Cover.jpg it show it as it should.

Only the new folder albums(with a capital letter) have this issue...

PS. I didn;t see your reply.

**Comment by Emilot** [comment]
![εικόνα](https://user-images.githubusercontent.com/18086896/197304360-e28388c3-d465-4fac-9fa9-aa1a9c1dc914.png)

i have this warnings...

**Comment by Emilot** [comment]
another one...more informative. This jpg opens just fine in Windows.

![εικόνα](https://user-images.githubusercontent.com/18086896/197305076-db2e0a88-944e-4856-a10c-1052acba616a.png)

as i see i was wrong at the point that it happens only with Capital letter, etc... Now i see some older albums that has the same issue and they have cover.jpg as it should...but it doesn't show.

**Comment by Emilot** [comment]
do you have a limitation in resolution?

![εικόνα](https://user-images.githubusercontent.com/18086896/197305338-0d7c0960-6aee-4cf9-ab8d-acb8b4919687.png)

this is not showing and it has Front.jpg as a filename.

No errors in debug log

**Comment by jcorporation** [comment, marked as answer]
**First screenshot: "invalid music directory value" error message**

Your mpd connection is not configured to use a socket. Therefore myMPD can not autodetect the music directory of MPD. The autoconfiguration works only with a socket connection. This is a MPD limitation of the "config" command https://mpd.readthedocs.io/en/latest/protocol.html#reflection. With no accessible music directory myMPD uses the "albumart" and "readpicture" commands from MPD. In MPD there is a hardcoded value of "cover" for coverimages (albumart command). The coverimage names setting has therefore no effect. This is documented here: https://jcorporation.github.io/myMPD/references/pictures


**Second screenshot:  "Could not determine mime type" error message**

myMPD can not detect the mime type of the image by the magic bytes. myMPD uses no full blown mime type detection library, but a own simply implementation. Your image is encoded as a special jpg type myMPD does not handle. This will be fixed.

> do you have a limitation in resolution?

There is a size limit of 5 MB for albumart retrieved through MPD.

**Comment by Emilot** [comment]
So as a conclusion i need to run MPD as socket and not as always under a specific path, etc... This changed after MPD 0.24 or it is myMPD change?

PS. Specify music path solved this issue :)

Thank you JC for your answer and explanation.

**Reply by jcorporation** [reply]
There was no change in MPD and myMPD. The `config` command is only available for MPD clients with a socket connection. You can keep the localhost:6600 connection, but you must set the music and playlist directory manually.

But a socket connection is the preferred way if MPD and myMPD runs on the same host. This avoids network overhead.

**Example:**
mpd.conf
```
# bind mpd to a unix socket
bind_to_address         "/run/mpd/socket"
```
myMPD connection settings:
![image](https://user-images.githubusercontent.com/765844/197356221-92101860-f7a0-4aba-8768-a04e4b0a0e86.png)

**Comment by Emilot** [comment]
Oh yeah i know about the socket thing, etc...and i prefer it but upmpdcli upnp client i use to my installation OS do not like it :(

I will take a deeper look at it, to ensure it.

PS. with the change i meant when this is changed because i always use localhost and myMPD(auto mode) never had a problem before...

**Reply by jcorporation** [reply]
You can enable both, unix socket and tcp/ip socket in the mpd.conf.
I use:
```
# For network
bind_to_address     "any"
bind_to_address     "/run/mpd/socket"
port               "6600"
```

> PS. with the change i meant when this is changed because i always use localhost and myMPD(auto mode) never had a problem before..

This behavior has not changed since I develop myMPD. 

myMPD 10.1.0 will show a warning for the not accessible music directory in the settings dialog.

**Comment by Emilot** [comment]
Oh i didn't know that :) Very useful... I 'll give it a try!!!

This is weird. I didn't change MPD configuration either and always run myMPD in "auto" mode. With 10.0.0 i get this error (cannot find mpd music directory)...but you must be right.
